detection_rules/rule.py

```diff
@@ -500,6 +501,17 @@ def get_asset(self) -> dict:
         """Generate the relevant fleet compatible asset."""
         return {"id": self.id, "attributes": self.contents.to_api_format(), "type": definitions.SAVED_OBJECT_TYPE}
 
+    def new(self, path: Optional[Path] = None, data: Optional[dict] = None, metadata: Optional[dict] = None):
```

Comment:
I don't personally like this `new` method. One advantage for the switch to dataclasses was type validation, and IMO this erases that. Is there any reason you can't do `dataclasses.replace` from the caller?

Reply:
I will revert all the references to this - good point